docs/advanced.rst

```diff
@@ -7,7 +7,9 @@ Custom HTTP client adaptor
 If for some reason you want to use neither aiohttp nor httpx, you can adapt your
 own HTTP client for usage with aiodynamo. To do so, create a class that conforms
 to the :py:class:`aiodynamo.http.base.HTTP` interface. Errors should be wrapped in
-a :py:exc:`aiodynamo.http.base.RequestFailed`.
+a :py:exc:`aiodynamo.http.base.RequestFailed`. When raising a :py:exc:`aiodynamo.http.base.RequestFailed`
+due to an exception in the HTTP client, you **should explicitly chain the exceptions**
+using ``raise RequestFailed(reason) from client_exception``.
```

Comment:
I'd rephrase the start of the paragraph, e.g.:

Aiodynamo comes with built-in support for `aiohttp` and `httpx` transports, and you can create your own by implementing the XX interface. 

Re explicitly chaining exceptions: please explain.
If it's only to help end-user debug errors, then I'd rather **recommend** to preserve context (or cause)?